### PR TITLE
Combine builders for `Vector.build` and `Vector.new_builder`

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Pull Request Description

We have decided to keep the old new_builder, and to combine the new and old builders into one builder.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso [hunk 1]
```diff
@@ -1579,6 +1345,9 @@ type Build_Style_Builder
          by the containing invocation of `Vector.build`.  The error is not added
          to the builder.
 
+         If an uncaught dataflow error is appended with `append`, the dataflow
+         error is propagated, and it is not added to the builder.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso]: Where is the error propagated? This is unclear to me.

Do you mean that the error will be propagated to the result of calling `to_vector`? Is the first such error propagated? Or last one?

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso]: Or looking at the code it seems that it is raised as a wrapped panic? That should be clarified by the docs. "Propagated" is far too vague here as it has several possible meanings.

**Review by radeusgd — APPROVED** [review]
I think that since we are keeping `new_builder`, we should have at least some tests of the error propagation, for example I think we should add a `new_builder` variant of the
```
group_builder.specify "Builder should propagate an uncaught dataflow error if it is appended with .append"
```
test

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso (hunk 1 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso]: Good point -- my documentation was written for `build` and didn't cover `new_builder`. I updated it to cover both cases, and added similar tests for `new_builder`. I changed 'propagate' to 'throw' where 'propagate' is confusing.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso [hunk 2]
```diff
@@ -179,8 +183,7 @@ type Vector a
     new_builder : Integer -> Builder
     new_builder (capacity=10) = Builder.new capacity
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso]: Why have the `new_builder` method at all, when we want to hide it? If users are supposed to use `Vector.build` in 99% of cases, then let's remove `new_builder` and keep just `Builder.new`! Copy this documentation to `Builder.new` then.

**Review by JaroslavTulach — APPROVED** [review]
I suggest to remove `Vector.new_builder`.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso]: Done.

**Pull request merged**